spectrum-network/src/network_controller.rs

```diff
@@ -171,6 +234,10 @@ where
                         conn_id: *conn_id,
                         enabled_protocols: HashMap::new(),
                     });
+                    // notify all handlers about new connection.
+                    for (_, ph) in self.supported_protocols.values() {
+                        ph.connected(*peer_id);
+                    }
                 }
                 ConnectedPeer::Connected { .. }
```

Comment:
If PM responsibility is "to manage connections - allocate/deallocate, confirm/deny" why NC sending CloseConnection without agreement with PM here?

Reply:
Bc here NC knows for sure that the connection already exists.

Reply:
By your design NC just syncronises PM and PCHs, but in this implementation NC contains some state related to PM. Is it correct? Because of it breaks design concepts

Reply:
Right, in order to synchronize PM with PCH and not create too much overhead on communication with those actors it has to track particular projections of peers state.